demisto_sdk/commands/pre_commit/pre_commit_command.py

```diff
@@ -138,68 +185,62 @@ def run(
             )
-            if python_version.startswith("2"):
-                with open(PRECOMMIT_PATH, "w") as f:
-                    yaml.dump(precommit_config, f)
-                if unit_test:
-                    response = subprocess.run(
-                        [
-                            sys.executable,
-                            "-m",
-                            "pre_commit",
-                            "run",
-                            "run-unit-tests",
-                            "-c",
-                            str(PRECOMMIT_PATH),
-                            "--files",
-                            *changed_files,
-                            "-v" if verbose else "",
-                        ],
-                        env=precommit_env,
-                        cwd=CONTENT_PATH,
-                    )
-                    if response.returncode:
-                        ret_val = response.returncode
-                continue
-            self.prepare_hooks(self._get_hooks(precommit_config), python_version)
-            with open(PRECOMMIT_PATH, "w") as f:
-                yaml.dump(precommit_config, f)
-            # use chunks because OS does not support such large comments
-            for chunk in more_itertools.chunked_even(changed_files, 10_000):
-                response = subprocess.run(
+
+        self.prepare_hooks(self._get_hooks(precommit_config))
+        if self.all_files:
+            precommit_config["exclude"] += f"|{join_files(exclude_files or {})}"
+        else:
+            precommit_config["files"] = join_files(self.files_to_run)
+        with open(PRECOMMIT_PATH, "w") as f:
+            yaml.dump(precommit_config, f)
+        if dry_run:
+            logger.info(
+                "Dry run, skipping pre-commit.\nConfig file saved to .pre-commit-config-content.yaml"
+            )
+            return ret_val
+        response = subprocess.run(
+            list(
+                filter(
+                    None,
                     [
                         sys.executable,
                         "-m",
                         "pre_commit",
                         "run",
+                        "-a",
                         "-c",
                         str(PRECOMMIT_PATH),
                         "--show-diff-on-failure" if show_diff_on_failure else "",
-                        "--files",
-                        *chunk,
                         "-v" if verbose else "",
                     ],
-                    env=precommit_env,
-                    cwd=CONTENT_PATH,
                 )
-                if response.returncode:
-                    ret_val = 1
+            ),
+            env=precommit_env,
+            cwd=CONTENT_PATH,
+        )
+        if response.returncode:
+            ret_val = 1
 
         # remove the config file in the end of the flow
         PRECOMMIT_PATH.unlink(missing_ok=True)
```

Comment:
if dry ron don't remove this file

Reply:
> if dry run don't remove this file

@ilaner In dry run, the pre-commit does not run at all, and only the file is saved, in line 214